ArcGISRuntimeSDKQt_CppSamples/Layers/BrowseWfsLayers/BrowseWfsLayers.cpp

```diff
@@ -0,0 +1,169 @@
+// [WriteFile Name=BrowseWfsLayers, Category=Geometry]
+// [Legal]
+// Copyright 2019 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+#include "BrowseWfsLayers.h"
+
+#include "Map.h"
+#include "MapQuickView.h"
+#include "WfsFeatureTable.h"
+#include "FeatureLayer.h"
+#include "QueryParameters.h"
+#include "SimpleRenderer.h"
+#include "SimpleLineSymbol.h"
+#include "SimpleMarkerSymbol.h"
+#include "WfsService.h"
+#include "WfsLayerInfo.h"
+
+using namespace Esri::ArcGISRuntime;
+
+BrowseWfsLayers::BrowseWfsLayers(QObject* parent /* = nullptr */):
+  QObject(parent),
+  m_map(new Map(Basemap::topographic(this), this))
+{
+  // create WFS Service
+  m_wfsService = new WfsService(QUrl("https://dservices2.arcgis.com/ZQgQTuoyBrtmoGdP/arcgis/services/Seattle_Downtown_Features/WFSServer?service=wfs&request=getcapabilities"),this);
+
+  // once WFS Service is loaded, populate QStringList with layer names for ComboBox
+  connect(m_wfsService, &WfsService::doneLoading, this, [this](Error e)
+  {
+    if (!e.isEmpty())
+      return;
+
+      m_wfsLayersInfoList = m_wfsService->serviceInfo().layerInfos();
+      for(auto i : m_wfsLayersInfoList)
```

Comment:
So, as good practice for samples, we try and reduce the amount of `autos` as much as possible to make explicit to the user what is going on. `auto` doesn't automatically add const-ref qualifiers either, so this is doing an implicit copy for every access! Try:

`for (const WfsLayerInfo& i : m_wfsLayersInfoList)`